### ISSUE TITLE
Add uninstall.php or uninstall routine

### ISSUE THREAD

**Issue by carstingaxion** [GatherPress/gatherpress#681]
### Is your enhancement related to a problem? Please describe.

Not that I want anybody to uninstall GatherPress, but people should be able to do that without any worries.

Following [WordPress best-practices for uninstallation](https://developer.wordpress.org/plugins/plugin-basics/uninstall-methods/), _GatherPress_ should have something like this somewhere.

```php
// if uninstall.php is not called by WordPress, die
if ( ! defined( 'WP_UNINSTALL_PLUGIN' ) ) {
    die;
}

$option_name = 'wporg_option';

delete_option( $option_name );

// for site options in Multisite
delete_site_option( $option_name );

// drop a custom database table
global $wpdb;
$wpdb->query( "DROP TABLE IF EXISTS {$wpdb->prefix}mytable" );
```
> In Multisite, looping through all blogs to delete options can be very resource intensive.

### Designs

Shouldn't be to much effort, if the existing & related methods can be re-used and we don't forget about the transients in #680 

https://github.com/GatherPress/gatherpress/blob/4942ce87071662cddcb8c993ae42893cad303830/includes/core/classes/class-setup.php#L196-L198

and 

https://github.com/GatherPress/gatherpress/blob/4942ce87071662cddcb8c993ae42893cad303830/includes/core/classes/class-setup.php#L307-L314

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by MervinHernandez** [comment]
✅ Viewed 681 - Uninstall workflows - Caleb has it

**Comment by MervinHernandez** [comment]
Hi @bluecollarcoders where do we stand on drafting an uninstall workflow?

**Comment by bluecollarcoders** [comment]
Hi @MervinHernandez 

I started work on the ticket. I apologize for the late response.

**Comment by mauteri** [comment]
It would be good to list all the data that needs to be removed, so @bluecollarcoders knows what the scope is. I'll look to get a list of everything some time this week and post it to the ticket. Thx!

**Comment by MervinHernandez** [comment]
#wcus_2024
@mauteri have we discussed - uninstall should yank `settings` only or should it also yank `data`?

**Comment by MervinHernandez** [comment]
✅  Answer = yank settings + data + custom tables

**Comment by mauteri** [comment]
I don't think this is a `good first issue` considering how destructive this can be. I think we can add something to our settings area (maybe Tools section) about what an uninstall will delete (opt-in checkboxes I'm thinking) so people aren't surprised with data that is removed.
